### ISSUE TITLE
Won't generate typescript definitions for 'components' -> 'headers' 

### ISSUE THREAD

**Issue by Piranit** [ferdikoomen/openapi-typescript-codegen#946]
### Problem.
I want to generate typescript definitions for the OpenAPI file. 
The generator doesn't generate typescript definitions for 'components' -> 'headers'.

### Expected behavior.
The generator generates typescript definitions for 'components' -> 'headers'.

### Steps to reproduce.
1. File: header-problem.v1.oas3.json
  ```
  {
    "openapi": "3.0.3",
    "info": {
      "title": "Header Problem",
      "version": "1.0.0"
    },
    "paths": {},
    "components": {
      "headers": {
        "Generic-Header": {
          "description": "Generic-Header description",
          "schema": { "type": "string" },
          "example": "123",
          "required": true
        }
      },
      "schemas": {
        "Generic-Schema": {
          "description": "Generic-Schema description",
          "type": "string",
          "example": "234",
          "required": true
        }
      }
    }
  }
  ```
2. run command `npx openapi-typescript-codegen --input ./header-problem.v1.oas3.json --output ./header-problem  --exportCore false --exportServices false`
3. It will generate typescript definitions for `Generic-Schema` but not for `Generic-Header`
   <img width="205" alt="Screenshot 2022-01-28 at 13 56 39" src="https://user-images.githubusercontent.com/8013207/151543272-cc6823d4-ce70-47ab-a599-682e6a728b52.png">
   <img width="1541" alt="Screenshot 2022-01-28 at 13 57 32" src="https://user-images.githubusercontent.com/8013207/151543284-4a4ca340-a8e3-4b68-87b6-de87793f3abc.png">

**Comment by ferdikoomen** [comment]
@Piranit Correct, the generator does not support this at the moment. How do you use the header? Normally these headers are referenced inside a Operation object, do you have an example of how you use them in your spec?

**Comment by Piranit** [comment]
@ferdikoomen based on documentation https://swagger.io/docs/specification/components/#:~:text=%23%20Reusable%20response%20headers,headers%3A we put all our reusable headers to `'components' -> 'headers'` section as we do for `'components' -> 'schemas'` for reusable schemas,  `'components' -> 'securitySchemes'` for reusable securitySchemes, etc.
We plan to have many OpenAPI Schemas and want to stay consistent between different API Schemas, so we have decided to create "Common API Schema Definitions". So the right place for all of our common headers is in the `'components' -> 'headers'` section.

Is it hard to include the `'components' -> 'headers'` section to typescript definitions generation?
